### ISSUE TITLE
Provide an SLA and a ticket system for the infrastructure maintained by the Foundation

### ISSUE THREAD

**Issue by mcollina** [openjs-foundation/security-collab-space#42]
The infrastructure provided by the OpenJS Foundation to the projects works great most of the time.
However, the Fastify website (https://fastify.io) was down for two days because of a DNS misconfiguration. The only person I know who has all access for us is @bensternthal, who was unavailable during the weekend for personal reasons.

We should provide more support to our projects and documented escalation paths to somebody in the Linux Foundation that manages this kind of critical infrastructure (e.g. DNS). Note that this burden should not fall only on @bensternthal, but it should be shared by a few people so that it's way more likely for somebody to be online to fix problems in case they arise during non-work hours.

**Comment by bensternthal** [comment]
@vvalderrv since the infa workstream is going to take some time I was wondering if we had an existing process that we could leverage in these circumstances. e.g. can community members just file[ jira tickets](https://jira.linuxfoundation.org/secure/Dashboard.jspa), if so do we have any documentation on how this works that I could review?

**Comment by tykeal** [comment]
While the Linux Foundation does not provide SLAs to projects, we do have a service desk available at https://support.linuxfoundation.org where any community member can raise issues related to systems and infrastructure that we are running for them. Escalations of issues are available through that system by way of a button that appears after a minimum of 1 hour of being open. Outside of business hours this escalation is supposed to alert our 24/7 on call staff of an issue that may need servicing on infrastructure that we do not have monitors on.

Now, while do do not provide SLAs, we do attempt to respond as quickly as possible during standard business hours.

Please note that we do not monitor the support queue outside of standard business hours, our 24/7 on call staff are responding alerts related to monitoring of infrastructure.

**Comment by tykeal** [comment]
As a side comment on the above issue, sounds like a change was made on or just before a weekend, LF Release Engineering and the Operations team strongly advise against this sort of activity as it can easily lead to problems such as what is described in the issue.

**Comment by mcollina** [comment]
> While the Linux Foundation does not provide SLAs to projects, we do have a service desk available at https://support.linuxfoundation.org/ where any community member can raise issues related to systems and infrastructure that we are running for them. Escalations of issues are available through that system by way of a button that appears after a minimum of 1 hour of being open. Outside of business hours this escalation is supposed to alert our 24/7 on call staff of an issue that may need servicing on infrastructure that we do not have monitors on.

@openjs-foundation/cpc I think we should document this somewhere in our docs.

> As a side comment on the above issue, sounds like a change was made on or just before a weekend, LF Release Engineering and the Operations team strongly advise against this sort of activity as it can easily lead to problems such as what is described in the issue.

@tykeal no change _should_ have been done to `fastify.io`: I did not request any. What was changed? The modification I asked to @bensternthal was about `fastify.dev`, which worked perfectly.

**Comment by ljharb** [comment]
I’m surprised an SLA wouldn’t be provided; that’s an industry standard that our projects should be able to rely on. If LF IT can’t provide one, perhaps we should find another infrastructure solution?

**Comment by bensternthal** [comment]
For me, the most important thing is that _support_  (broadly defined) is in scope of the infra work the LF IT team is doing as part of the Sovereign Tech Fund work. AFAIK it will be. Let's give LF IT a chance to do their research, interview projects, and propose informed solutions.

**Comment by ljharb** [comment]
That's great - but I think a critical piece is beginning to provide an SLA moving forward.

**Comment by ljharb** [comment]
@tykeal I appear to have had a misunderstanding - you may have meant SLA in terms of responsiveness, while I was thinking SLA in terms of website/service uptime? Either way, I hope that we can figure out a way to get everyone's needs met, since obviously, sharing and utilizing LF IT's resources is the most efficient path, and I look forward to working with LF IT to figure this out!

**Comment by tykeal** [comment]
@ljharb you're correct SLA does translate to a general responsiveness agreement in every IT department I've ever worked in ;)

Again, LFIT does not provide such SLAs to our projects. The reason being we just aren't staffed for that sort of support and it would dramatically increase our costs if we were! We do, however offer our support queue which is staffed by our operations staff, release engineering staff, and our LFX platform support staff. We strive for, but do not guarantee, a response (but not necessarily resolution) with in 4 hours during standard business hours. Outside of that we have 24/7 monitoring of infrastructure that we have direct control over, that is we're in charge of the actual system that hosts it.

**Comment by mcollina** [comment]
Given that LFIT cannot provide SLAs for OpenJS, can access to critical system (eg DNS or even servers) be shared with more than one person in OpenJS (currently @bensternthal has access)? Ultimately problems would very seldomly arise, but when they do _it's a big issue_ and the maintainers are _personally_ grilled because things stop working.
